### PR TITLE
Add TLS support for proxytest

### LINKED ISSUE #4497: Extend `testing/proxytest` to support mTLS
**Describe the enhancement:**

Extend the `testing/proxytest` to support mTLS. It should be possible to configure:
 - the CA the proxy will use for its own requests to other servers
 - the CA to validate the clients' TLS certificate
 - the TLS certificate to present to its clients
 - the key for the certificate presented to its clients

**Describe a specific use case for the enhancement or feature:**

Test the Elastic Agent support for mTLS between itself and fleet-server or a proxy. More specifically, to do an integration test for test
 - https://github.com/elastic/elastic-agent/issues/2247
 - https://github.com/elastic/elastic-agent/issues/2248

**What is the definition of done?**

cURL can use the testproxy when it requires mTLS. The request will only succeed if cURL is correctly configured to present TLS certificates and with the proxy's CA.

The proxy is configured with:
 - cURL CA's
 - proxy-cert.pem
 - procy-cert-key.pem

cURL is used to reach https://elastic.co as follows:
```
curl --cacert proxy-ca.pem \
     --cert cRUL-cert.pem \
     --key cRUL-cert-key.pem \
     https://elastic.co
```

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?

This PR introduces TLS configuration for `testing/proxytest` in order to be able to configure mTLS.
No TLS, basic TLS (certificate server-side, no client auth) and mTLS (certificates for both client and server, common trusted CA) are tested via unit tests

## Why is it important?
`testing/proxytest` is going to be used to test (m)TLS proxy settings for `elastic-agent`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Related issues

- Closes #4497 

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: testing/proxytest/proxytest.go [hunk 1]
```diff
@@ -143,6 +154,32 @@ func New(t *testing.T, optns ...Option) *Proxy {
 	return &p
 }
 
+func (p *Proxy) Start() {
+	p.Server.Start()
+	u, err := url.Parse(p.URL)
+	if err != nil {
+		panic(fmt.Sprintf("could parse fleet-server URL: %v", err))
```
Comment (blakerouse) [inline testing/proxytest/proxytest.go:161]: Why not make this return an error and let the caller handle it?

File: testing/proxytest/proxytest.go [hunk 2]
```diff
@@ -143,6 +154,32 @@ func New(t *testing.T, optns ...Option) *Proxy {
...
+	p.LocalhostURL = "http://localhost:" + p.Port
+
+	p.opts.logFn("running on %s -> %s", p.URL, p.LocalhostURL)
+}
+
+func (p *Proxy) StartTLS() {
+	p.Server.StartTLS()
+	u, err := url.Parse(p.URL)
+	if err != nil {
+		panic(fmt.Sprintf("could parse fleet-server URL: %v", err))
```
Comment (blakerouse) [inline testing/proxytest/proxytest.go:174]: Same here.

**Review by blakerouse — APPROVED** [review]
Looks good. Just a questions on the panic not a blocker.

File: testing/proxytest/proxytest.go (hunk 1 above)
Comment (pchila) [reply testing/proxytest/proxytest.go:161]: Originally the Start() and StartTLS() methods were not defined for the `Proxy` struct as it exposed the ones from `*httptest.Server`. It wasn't even used in tests as the server was created and started in one go.
Now that we are overriding those methods we can return what we want, will add a small commit for that

Comment (pchila) [reply testing/proxytest/proxytest.go]: Done in 979217c0b27b525e39d2344354e5e2e6363198b4

File: testing/proxytest/proxytest.go (hunk 2 above)
Comment (pchila) [reply testing/proxytest/proxytest.go]: Done in 979217c0b27b525e39d2344354e5e2e6363198b4

**Pull request merged**